scipy/linalg/tests/test_matfuncs.py

```diff
@@ -1045,3 +1048,13 @@ def test_empty(self):
         b = np.empty((5, 0))
         res = khatri_rao(a, b)
         assert_allclose(res, np.empty((15, 0)))
+
+@pytest.mark.parametrize('func', [cosm, sinm, tanm, coshm, sinhm, tanhm])
+@pytest.mark.parametrize('dt', [int, float, np.float32, complex, np.complex64])
+def test_empty_matrix_input(func, dt):
+    A = np.zeros((0, 0), dtype=dt)
+    result = func(A)
+    A0 = np.zeros((1, 1), dtype=dt)
```

Comment:
As above, let's make this to a larger number to ensure that we compare against the general case rather than a special case.